eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -80,223 +72,153 @@ class DoctrineDatabase extends Gateway
+                        ParameterType::INTEGER
+                    ),
+                    'created' => $query->createPositionalParameter(
+                        $versionInfo->creationDate,
+                        ParameterType::INTEGER
+                    ),
+                    'status' => $query->createPositionalParameter(
+                        $versionInfo->status,
+                        ParameterType::INTEGER
+                    ),
+                    'initial_language_id' => $query->createPositionalParameter(
+                        $this->languageHandler->loadByLanguageCode(
+                            $versionInfo->initialLanguageCode
+                        )->id,
+                        ParameterType::INTEGER
+                    ),
+                    'contentobject_id' => $query->createPositionalParameter(
+                        $versionInfo->contentInfo->id,
+                        ParameterType::INTEGER
+                    ),
+                    'language_mask' => $query->createPositionalParameter(
+                        $this->languageMaskGenerator->generateLanguageMaskForFields(
+                            $fields,
+                            $versionInfo->initialLanguageCode,
+                            $versionInfo->contentInfo->alwaysAvailable
+                        ),
+                        ParameterType::INTEGER
+                    ),
+                ]
+            );
 
-        $q->prepare()->execute();
+        $query->execute();
 
-        return (int)$this->dbHandler->lastInsertId(
-            $this->dbHandler->getSequenceName('ezcontentobject_version', 'id')
-        );
+        return (int)$this->connection->lastInsertId(self::CONTENT_VERSION_SEQ);
     }
 
-    /**
-     * Updates an existing content identified by $contentId in respect to $struct.
-     *
-     * @param int $contentId
-     * @param \eZ\Publish\SPI\Persistence\Content\MetadataUpdateStruct $struct
-     * @param \eZ\Publish\SPI\Persistence\Content\VersionInfo $prePublishVersionInfo Provided on publish
-     *
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
-     */
-    public function updateContent($contentId, MetadataUpdateStruct $struct, VersionInfo $prePublishVersionInfo = null)
-    {
+    public function updateContent(
+        int $contentId,
+        MetadataUpdateStruct $struct,
+        ?VersionInfo $prePublishVersionInfo = null
+    ): void {
         $query = $this->connection->createQueryBuilder();
-        $query->update('ezcontentobject');
+        $query->update(self::CONTENT_ITEM_TABLE);
 
         $fieldsForUpdateMap = [
             'name' => ['value' => $struct->name, 'type' => ParameterType::STRING],
-            'initial_language_id' => ['value' => $struct->mainLanguageId, 'type' => ParameterType::INTEGER],
+            'initial_language_id' => [
+                'value' => $struct->mainLanguageId,
+                'type' => ParameterType::INTEGER,
+            ],
```

Comment:
> IMHO all `$fieldsForUpdateMap` items should be formatted in this way or none of them

Hmm, single line felt more readable. This one item needed to be like that because it exceeded max line length and got re-formatted by PhpStorm.
But ok - gonna change to multiline, it doesn't look that bad actually.

_Note: when working with Content Type GW I decided to skip assoc array entirely here for readability, sorry. Can align this one, but don't want to make too much fuss ;)_

Reply:
Reformatted and squashed it into 8d42846.